python/lsst/daf/butler/registry/dimensions/table.py

```diff
@@ -60,12 +62,25 @@ def clearCaches(self):
         # Docstring inherited from DimensionRecordStorage.clearCaches.
         pass
 
-    def getElementTable(self, dataId: Optional[DataId] = None) -> Optional[sqlalchemy.sql.FromClause]:
-        return self._table
-
-    def getCommonSkyPixOverlapTable(self, dataId: Optional[DataId] = None
-                                    ) -> Optional[sqlalchemy.sql.FromClause]:
-        return None
+    def join(
+        self,
+        builder: QueryBuilder, *,
+        regions: Optional[NamedKeyDict[DimensionElement, sqlalchemy.sql.ColumnElement]] = None,
+        timespans: Optional[NamedKeyDict[DimensionElement, Timespan[sqlalchemy.sql.ColumnElement]]] = None,
+    ):
+        assert regions is None, "This implementation does not handle spatial joins."
+        joinDimensions = list(self.element.graph.required)
+        joinDimensions.extend(self.element.implied)
+        joinOn = builder.startJoin(self._table, joinDimensions, self.element.RecordClass.__slots__)
```

Comment:
Im slightly uncomfortable directly accessing __slots__ here, though I dont think it can hurt....

Reply:
I've at least [documented](https://pipelines.lsst.io/v/daily/py-api/lsst.daf.butler.DimensionRecord.html#lsst.daf.butler.DimensionRecord) that `__slots__` is considered part of the public API for `DimensionRecord`, but I have always wondered whether it'd be better to add a differently-named class attribute with the exact same elements instead.  If you have a strong preference for that, this would not be a bad ticket to make that change.